tools/compare_benchmark.py

```diff
@@ -0,0 +1,159 @@
+def run_benchmark(branch_or_commit: str, benchmark_args: str, out_dir: str) -> str:
+    benchmark_out = os.path.join(out_dir, branch_or_commit + ".json")
+    if os.path.exists(benchmark_out):
+        print(f"{benchmark_out} already exists. Skip benchmarking {branch_or_commit}.")
+        return benchmark_out
+
+    subprocess.check_call(f"git checkout {branch_or_commit}", shell=True)
+
+    subprocess.check_call("pip install -e .", shell=True)
+
+    benchmark_command = " ".join(
+        (
+            "bin/nvfuser_bench",
+            benchmark_args,
+            f"--benchmark_out={benchmark_out}",
+            "--benchmark_format=json",
+        )
+    )
+    print("Running benchmark command: " + benchmark_command)
+    with open(os.path.join(out_dir, branch_or_commit + ".stdout"), "w") as stdout, open(
+        os.path.join(out_dir, branch_or_commit + ".stderr"), "w"
+    ) as stderr:
+        subprocess.check_call(
+            benchmark_command, stdout=stdout, stderr=stderr, shell=True
+        )
+    print(f"The benchmark output is stored in {benchmark_out}.")
+
+    return benchmark_out
+
+
+@dataclass
+class Comparison:
+    name: str
+    baseline_time: float
+    contender_time: float
+    # contender_time divided by baseline_time. Smaller is better.
+    change: float
+    time_unit: str
+
+    def __str__(self):
+        return f"Benchmark {self.name} changed from {self.baseline_time}{self.time_unit} to {self.contender_time}{self.time_unit} ({self.change:.2f}x)"
+
+
+# Compares the two given benchmark results and produces a .json file containing
+# the comparison.
+def compare(baseline_out: str, contender_out: str, out_dir: str) -> str:
+    baseline, _ = os.path.splitext(os.path.basename(baseline_out))
+    contender, _ = os.path.splitext(os.path.basename(contender_out))
+    comparison_out = os.path.join(out_dir, f"{baseline}_vs_{contender}.json")
+    subprocess.check_call(
+        f"third_party/benchmark/tools/compare.py -d {comparison_out} benchmarks {baseline_out} {contender_out}",
+        shell=True,
+    )
+    return comparison_out
+
+
+def load_comparison(comparison_out: str) -> list[Comparison]:
+    comparisons: list[Comparison] = []
+    with open(comparison_out) as f:
+        data = json.loads(f.read())
+
+    for row in data:
+        for measurement in row["measurements"]:
+            if row["run_type"] != "iteration":
+                continue
+            comparisons.append(
+                Comparison(
+                    row["name"],
+                    measurement["real_time"],
+                    measurement["real_time_other"],
+                    measurement["time"] + 1,
+                    row["time_unit"],
+                )
+            )
+
+    return comparisons
+
+
+def summarize_comparison(comparisons: Iterable[Comparison], out_dir: str) -> None:
+    sorted_comparisons = sorted(comparisons, key=lambda x: x.change)
+
+    # Print top improvements and regressions.
+    num_tops = 5
```

Comment:
It doesn't necessarily have to happen in this PR but it might be useful to split this function into its own script and expose some of these params as cmdline args. That way we can do the expensive benchmarking then adjust plots afterward.

Reply:
I mentioned this around `run_benchmark`. This script skips benchmarking when `out_dir/branch_or_commit.json` already exists. So the user can copy benchmark results to a folder before running this script to skip the expensive benchmarking.